webapp/channels/src/components/post_view/post_image/post_image.tsx

```diff
@@ -23,46 +26,55 @@ interface Props {
     };
 }
 
-export default class PostImage extends React.PureComponent<Props> {
-    showModal = (e: (KeyboardEvent<HTMLImageElement> | MouseEvent<HTMLElement>), link = '') => {
+const PostImage = ({
+    imageMetadata,
+    link,
+    post,
+    actions,
+}: PostImageProps) => {
+    const showModal = useCallback((
+        e: KeyboardEvent<HTMLImageElement> | MouseEvent<HTMLElement>,
+        link = '',
+    ) => {
         e.preventDefault();
 
-        this.props.actions.openModal({
+        actions.openModal({
             modalId: ModalIdentifiers.FILE_PREVIEW_MODAL,
             dialogType: FilePreviewModal,
             dialogProps: {
-                post: this.props.post,
+                post,
                 startIndex: 0,
-                fileInfos: [{
-                    has_preview_image: false,
-                    link,
-                    extension: this.props.imageMetadata.format,
-                    name: this.props.link,
-                }],
+                fileInfos: [
+                    {
+                        has_preview_image: false,
+                        link,
+                        extension: imageMetadata.format,
+                        name: link,
+                    },
+                ],
             },
         });
-    };
+    }, [actions, imageMetadata.format, post]);
```

Comment:
Although i believe we have one single `openModal` in action object, we should try to avoid large objects as dependencies. i would recommend to de-structured the openModal and pass it as dependency here to be explicit